tsl/test/expected/bgw_scheduler_restart.out

```diff
@@ -0,0 +1,85 @@
+-- This file and its contents are licensed under the Timescale License.
+-- Please see the included NOTICE for copyright information and
+-- LICENSE-TIMESCALE for a copy of the license.
+\c :TEST_DBNAME :ROLE_SUPERUSER
+CREATE VIEW tsdb_bgw AS
+       SELECT datname, pid, backend_type, application_name
+         FROM pg_stat_activity
+        WHERE application_name LIKE '%TimescaleDB%'
+     ORDER BY datname, backend_type, application_name;
+-- Show the default scheduler restart time
+SHOW timescaledb.bgw_scheduler_restart_time;
+ timescaledb.bgw_scheduler_restart_time 
+----------------------------------------
+ 1min
+(1 row)
+
+SELECT _timescaledb_functions.start_background_workers();
+ start_background_workers 
+--------------------------
+ t
+(1 row)
+
+SELECT pg_sleep(10);		-- Wait for scheduler to start.
+ pg_sleep 
+----------
+ 
+(1 row)
+
+SELECT datname, application_name FROM tsdb_bgw;
+         datname          |            application_name             
+--------------------------+-----------------------------------------
+ db_bgw_scheduler_restart | TimescaleDB Background Worker Scheduler
+                          | TimescaleDB Background Worker Launcher
+(2 rows)
+
+ALTER SYSTEM SET timescaledb.shutdown_bgw_scheduler TO 'on';
+ALTER SYSTEM SET timescaledb.debug_bgw_scheduler_exit_status TO 1;
+SELECT pg_reload_conf();
+ pg_reload_conf 
+----------------
+ t
+(1 row)
+
+SELECT pg_sleep(20);		-- Wait for scheduler to exit.
+ pg_sleep 
+----------
+ 
+(1 row)
+
+SELECT datname, application_name FROM tsdb_bgw;
+ datname |            application_name            
+---------+----------------------------------------
+         | TimescaleDB Background Worker Launcher
+(1 row)
+
+ALTER SYSTEM RESET timescaledb.shutdown_bgw_scheduler;
+ALTER SYSTEM RESET timescaledb.debug_bgw_scheduler_exit_status;
+SELECT pg_reload_conf();
+ pg_reload_conf 
+----------------
+ t
+(1 row)
+
+SELECT pg_sleep(60);		-- Wait for scheduler to restart.
```

Comment:
These sleeps seem to add 80 seconds (1+ minute) to the test suite. That is quite a lot for one test. Also, sleeps for synchronization is inherently flaky so it would be good to come up with a solution where the processes could synchronize, e.g., around a lock or similar? 

Is there maybe a way you can wait on a pid?

Reply:
Internally there's at least `WaitForBackgroundWorkerStartup` and `WaitForBackgroundWorkerShutdown`. Can't we expose them via SQL functions to avoid sleeping?